packages/jest-mock/src/index.js

```diff
@@ -274,6 +248,53 @@ class ModuleMockerClass {
     this._invocationCallCounter = 1;
   }
 
+  _getSlots(object?: Object): Array<string> {
+    if (!object) {
+      return [];
+    }
+
+    const slots = new Set();
+    const EnvObjectProto = this._environmentGlobal.Object.prototype;
+    const EnvFunctionProto = this._environmentGlobal.Function.prototype;
+    const EnvRegExpProto = this._environmentGlobal.RegExp.prototype;
+
+    // Also check the builtins in the current context as they leak through
+    // core node modules.
+    const ObjectProto = Object.prototype;
+    const FunctionProto = Function.prototype;
+    const RegExpProto = RegExp.prototype;
+
+    // Properties of Object.prototype, Function.prototype and RegExp.prototype
+    // are never reported as slots
+    while (
+      object != null &&
+      object !== EnvObjectProto &&
+      object !== EnvFunctionProto &&
+      object !== EnvRegExpProto &&
+      object !== ObjectProto &&
+      object !== FunctionProto &&
+      object !== RegExpProto
```

Comment:
Create  a `Set` instead and verify that `object` ∉ `blacklist`.

Reply:
It considered that but the number of cases is still manageable and `Set`s are 3x slower in this case.